### ISSUE TITLE
Alignment of languages in dropdown

### ISSUE THREAD

**Issue by share-with-me** [goavki/apertium-html-tools#127]
In the current version, all the languages (whether `ltr` or `rtl` appear all on one end - either left or right depending on the `html dir` of the page. Would it make more sense if all the `ltr` lanaguages appear on left end of dropdown (irrespective of language) and similar for `rtl` languages? The following screenshots describe the issue in more detail:

**Current Version**
![screen shot 2017-03-24 at 1 29 33 am](https://cloud.githubusercontent.com/assets/8894636/24267783/735fc2a4-1031-11e7-9990-a0bb6799e5ba.png)

**Proposed Version** - colors for highlighting
![screen shot 2017-03-24 at 1 24 13 am](https://cloud.githubusercontent.com/assets/8894636/24267795/7fb0bb26-1031-11e7-9a0a-7933b810aeb2.png)
As we can see, the **`rtl` languages (in red)** appear from **right to left** in the dropdown.

Thus, the langauges in the dropdown should be independent of the parent html dir and follow its own dir on any `index.lang.html` page. 
Does this feature make sense semantically?

**Comment by sushain97** [comment]
Yes, this makes sense and has been brought up before. It isn't possible with the standard `<select>` element.

**Comment by share-with-me** [comment]
> It isn't possible with the standard `<select>` element.

Umm, I tried to target the js code where these options for dropdown get generated and applied the CSS there based on whether the langauge is `rtl` or `ltr`. Should I make a gist? I do not completely get what you meant by that?

> Nice catch though

Thank you so much :D

**Comment by sushain97** [comment]
> Umm, I tried to target the js code where these options for dropdown get generated and applied the CSS there based on whether the langauge is rtl or ltr. Should I make a gist? I do not completely get what you meant by that?

The behavior that you describe in the second screenshot is not possible using the HTML `<select>` element as far as I am aware. Did it work for you?

**Comment by sushain97** [comment]
Does your solution work on Chrome and Firefox?

**Comment by share-with-me** [comment]
On firefox, it is working. Did not check on chrome! Would update you once I check it! 
Off-topic: I have sent you an email regarding the ideas. Please have a look! Thanks :D . I will edit this comment.

**Comment by share-with-me** [comment]
@sushain97 I tried on Firefox and Chrome. It does appear styled on Firefox but fails to appear that way in Chrome. On googling, the results show that it is not possible to style the `select option` elements in the browsers that are `webkit-based` such as Chrome and Safari. This is a [stackoverflow link](http://stackoverflow.com/questions/12301729/styling-option-value-in-select-drop-down-html-not-work-on-chrome-safari) for the same argument. I am attaching the screenshots. 

**Firefox**
![screen shot 2017-03-24 at 11 56 53 am](https://cloud.githubusercontent.com/assets/8894636/24283086/a77ac98c-1089-11e7-8194-eaa08b020d55.png)

**Chrome**
![screen shot 2017-03-24 at 11 57 24 am](https://cloud.githubusercontent.com/assets/8894636/24283092/aeae030e-1089-11e7-8812-4591c90275c5.png)

**Comment by share-with-me** [comment]
@sushain97 I have tried to work around the above issue by making use of a `bootstrap dropdown` instead of `select option`. The advantage of this will be freedom to style the dropdown `li` elements any way we want; which is what I have tried to accomplish. Works well on both Firefox and Chrome. Here is the [link for the demo](https://drive.google.com/open?id=0BwrSYlOsMCyWM1lILUUtUW0tY28) .
Please have a look and review if this makes progress towards solving of issue.
Thanks!

**Comment by sushain97** [comment]
> On googling, the results show that it is not possible to style the select option elements in the browsers that are webkit-based such as Chrome and Safari. 

This problem is what I was alluding to earlier.

> Please have a look and review if this makes progress towards solving of issue.

This is a good start. The styling needs work but otherwise this might work. On mobile, we must revert to a normal `select` element.

@jonorthwash may have suggestions here as well.

**Comment by share-with-me** [comment]
> This problem is what I was alluding to earlier.

Exactly. It worked on Firefox and I assumed it would work well on Chrome and Safari too. But googling made me wiser :P .

> This is a good start. The styling needs work but otherwise this might work. On mobile, we must revert to a normal `select` element.

Thank you :D. Yepp, the styling does need work but I was trying to make a dropdown of `li` so that we can have full freedom about how to style it. Yes, we could revert to `select` only in case of mobiles. I would love to work on this after discussing ideas with you and @jonorthwash  :D

**Comment by jonorthwash** [comment]
Duplicate of #19.

**Comment by share-with-me** [comment]
@jonorthwash , oh cool! did not see this previously! Umm, does this workaround try to achieve what we are trying (like alignment of `rtl` and `ltr`) languages in the dropdown? As @sushain97 said, we could implement this on laptops and bigger screens and revert to our good old `select` while on mobile and tablets. Also, should I close this issue here? and resume on #19 ?
Thanks!

**Comment by jonorthwash** [comment]
Not sure.  @sushain97 ?

**Comment by sushain97** [comment]
I believe these are separate issues. #19 addressed having the languages all be either right-aligned or left-aligned. This issue seeks to make them both right-aligned and left-aligned, whichever the preference of the language is. In a sense, this would make it so the current interface language does not effect the alignment of the language options. @jonorthwash what would you see as the right behavior?

**Comment by jonorthwash** [comment]
> whichever the preference of the language is

You mean the language whose name is listed, not the language  the page is localised in, right?

> what would you see as the right behavior?

I would want the language names to be aligned per the script of the language name.  So e.g., Uyghur and Hebrew (as we have them localised currently) would be right-aligned, but Mandarin and Kazakh (as we have them localised currently) would be left-aligned.

**Comment by sushain97** [comment]
@jonorthwash 

> You mean the language whose name is listed, not the language the page is localised in, right?

Precisely.

> I would want the language names to be aligned per the script of the language name. So e.g., Uyghur and Hebrew (as we have them localised currently) would be right-aligned, but Mandarin and Kazakh (as we have them localised currently) would be left-aligned.

OK. This is contrary to the current behavior (see apertium.org) and indeed what @share-with-me has presented. Correct?

**Comment by share-with-me** [comment]
@jonorthwash  As @sushain97 said, this fix tries to make the languages in the dropdown option next to the globe. My issue was to make the languages within that display dropdown as left-aligned or right-aligned depending only on the rtl or ltr behaviour of that language and not making it dependent on the page that is localised in.
